app/views/accounts/new.html.erb

```diff
@@ -64,7 +64,9 @@
       <%= link_to new_account_path(step: 'method', type: params[:type]), class: "flex w-8 h-8 shrink-0 grow-0 items-center justify-center rounded-lg bg-[#141414]/5" do %>
         <%= lucide_icon('arrow-left', class: 'text-gray-500 w-5 h-5') %>
       <% end %>
-      <span>Add account</span>
+      <% if params[:type].present? && Account.accountable_types.include?("Account::#{params[:type]}") %>
+        <span>Add <%= t('.'+params[:type]) %></span>
```

Comment:
The whole string within span ought to be captured for internationalisation, and then the model's name passed down as a variable.

You really need access to the constantized account type though, which is where it gets more tricky. That should probably be done in the controller.

Assuming the accountable type was made available you could do something like:

```ruby
t('.new', account_type: @accountable_type.model_name.human.downcase)
```

Then in the locale file:
```
account_type: "New %{account_type}"
```

Reply:
Sounds good. Thanks for the explanation!